ctypes.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Purego only works on linux/macOS with amd64 and arm64 from now
+//go:build (linux || darwin) && (amd64 || arm64) && !cgo
+
+package waf
+
+import "unsafe"
+
+const (
+	wafMaxStringLength   = 4096
+	wafMaxContainerDepth = 20
+	wafMaxContainerSize  = 256
+	wafRunTimeout        = 5000
+)
+
+type wafReturnCode int32
+
+const (
+	wafErrInternal        wafReturnCode = -3
+	wafErrInvalidObject                 = -2
+	wafErrInvalidArgument               = -1
+	wafOK                               = 0
+	wafMatch                            = 1
+)
+
+// wafObjectType is an enum in C which has the size of DWORD.
+// But DWORD is 4 bytes in amd64 and arm64 so uint32 it is.
+type wafObjectType uint32
+
+const (
+	wafInvalidType wafObjectType = 0
+	wafIntType                   = 1 << 0
+	wafUintType                  = 1 << 1
+	wafStringType                = 1 << 2
+	wafArrayType                 = 1 << 3
+	wafMapType                   = 1 << 4
+)
+
+type wafObject struct {
+	parameterName       uintptr
+	parameterNameLength uint64
+	value               uintptr
+	nbEntries           uint64
+	_type               wafObjectType
+	_                   [4]byte // Forced padding
```

Comment:
We only support 2 archs and cgo generated the same padding so yes. No because actually go will do the same padding itself, we juste add it explicitly to not take any change. And we cannot pack a struct in go so it will get tricky if the struct is packed (apart from breaking all tracers of course)